src/main/java/io/kestra/plugin/weaviate/BatchCreate.java

```diff
@@ -33,29 +41,81 @@
             code = {
                 "host: localhost:8080",
                 "apiKey: some_api_key",
-                "className: WeaviateObject"
+                "className: WeaviateObject",
+                "objects: " +
+                "   textField: \"text\"",
+                "   numField: 32"
             }
         )
     }
 )
-public class BatchCreate extends WeaviateConnection implements RunnableTask<BatchCreate.Output>, BatchCreateInterface {
-
-    protected String className;
+public class BatchCreate extends WeaviateConnection implements RunnableTask<BatchCreate.Output> {
+
+    @Schema(
+        title = "Class name where you want to insert data"
+    )
+    @PluginProperty(dynamic = true)
+    private String className;
+
+    @Schema(
+        title = "Objects to create with their properties",
+        description = "ION File URI or the list of objects to insert",
+        anyOf = {
+            String.class,
+            Map[].class,
+            List.class,
+        }
+    )
+    @NotNull
+    private Object objects;
 
-    @Builder.Default
-    protected List<Map<String, Object>> parameters = List.of(new HashMap<>());
+    private StorageInterface storageInterface;
 
     @Override
     public BatchCreate.Output run(RunContext runContext) throws Exception {
         WeaviateClient client = connect(runContext);
+        Optional<URI> uri = Optional.empty();
+
+        List<WeaviateObject> weaviateObjects = new ArrayList<>();
+
+        if (objects instanceof List objectList) {
+            if (!objectList.isEmpty() && objectList.get(0) instanceof Map<?,?>) {
+                objectList.stream()
+                    .map(param -> WeaviateObject.builder()
+                        .id(UUID.randomUUID().toString())
+                        .className(className)
+                        .properties((Map<String, Object>) param)
+                        .build())
+                    .forEach(object -> weaviateObjects.add((WeaviateObject) object));
+            }
+        } else if (objects instanceof Map[]) {
+            Arrays.stream((Map<String, Object>[]) objects)
+                .map(param -> WeaviateObject.builder()
+                    .id(UUID.randomUUID().toString())
+                    .className(className)
+                    .properties(param)
+                    .build())
+                .forEach(weaviateObjects::add);
+        }
```

Comment:
It will never happen, see what was done [here](https://github.com/kestra-io/plugin-aws/commit/c2b188c3c2fdaff2b56630d2d22176c7fdc517ec#diff-668ca875546bf9c9f99bf1bb34686d4c7fa1517151dbd1cc2e9da82647e085fdR161-R178): you get either a List or a String (and if it's a list you can cast items to Map as you expect Map[] which is converted to List<Map> by Jackson)